### PR TITLE
69 introduce typescript

### LINKED ISSUE #69: Introduce TypeScript
Since solid-ui is a library that _is_ used by multiple packages, it makes sense to provide type definitions, to make it easier for consumers to figure out how to use it.

### PR DESCRIPTION
**Pull request by Vinnl**
Builds on #68. This keeps compiling files with Babel like before, but additionally runs `tsc` to generate `.d.ts` files for TypeScript files.

I believe this should work just like before, but... I can't load my local copy of solid-ui anymore, whether this branch or `master`, so I can't really test it :(

Fixes #69.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -48,21 +54,23 @@ const UI = {
   create: require('./create'),
   icons: require('./iconBase'),
   log: require('./log'),
-  matrix: require('./matrix'),
+  matrix: matrix,
```
Comment (megoth) [inline src/index.ts]: Maybe just use the short hand form, to clean up the code up a bit more?

File: src/preferences.ts [hunk 2]
```diff
@@ -78,42 +86,47 @@ function recordPersonalDefaults (klass, context) {
...
-    }, err => {
+    }, (err: Error) => {
       reject(err)
     })
   })
 }
 
-function renderPreferencesForm (subject, klass, preferencesForm, context) {
+function renderPreferencesForm (
+  subject: $rdf.NamedNode,
```
Comment (megoth) [inline src/preferences.ts]: Isn't it better with simply `NamedNode`?

File: src/index.ts (hunk 1 above)
Comment (Vinnl) [reply src/index.ts]: I'm generally not a fan of mixing the shorthand and regular form (or of using the shorthand form outside of one-liners in general), but I don't feel strongly about it so if you do - sure :) In that case, might also be a good idea to add an ESLint rule?

File: src/preferences.ts (hunk 2 above)
Comment (Vinnl) [reply src/preferences.ts]: Haven't heard of that before, why? Should I change the other `$rdf.*` imports as well?

**Comment by megoth** [comment]
Closing this in favor of https://github.com/solid/solid-ui/pull/127

**Pull request closed without merging**